src/test/java/com/softserve/academy/spaced/repetition/service/CourseCommentServiceTest.java

```diff
@@ -0,0 +1,109 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@ActiveProfiles("testdatabase")
+@SpringBootTest
+@Import(TestDatabaseConfig.class)
+@Sql("/data/TestData.sql")
+@Transactional
+public class CourseCommentServiceTest {
+
+    private static final long COURSE_ID = 1L;
+
+    private CourseCommentService courseCommentServiceUnderTest;
+
+    @Autowired
+    private CourseCommentRepository courseCommentRepository;
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Mock
+    private UserService mockedUserService;
+
+    @Mock
+    private CommentFieldsValidator mockedCommentFieldsValidator;
+
+    @Before
+    public void setUp() throws Exception {
+        courseCommentServiceUnderTest = new CourseCommentService(courseCommentRepository, courseRepository, mockedUserService, mockedCommentFieldsValidator);
+    }
+
+    private User createMockedUser() {
+        User mockedUser = new User(new Account("user@email.com"), new Person("Ivan", "Kruk"), new Folder());
+        mockedUser.getPerson().setId(1l);
+        return mockedUser;
+    }
+
+    @Test
+    public void testAddCommentForCourse() throws Exception {
+        when(mockedUserService.getAuthorizedUser()).thenReturn(createMockedUser());
+        CourseComment savedComment = courseCommentServiceUnderTest.addCommentForCourse("Very interesting", COURSE_ID);
+        assertNotNull(savedComment);
+    }
+
+    @Test
+    public void testGetCommentById() throws Exception {
+        when(mockedUserService.getAuthorizedUser()).thenReturn(createMockedUser());
+        CourseComment savedComment = courseCommentServiceUnderTest.addCommentForCourse("Very interesting", COURSE_ID);
+        assertNotNull(courseCommentServiceUnderTest.getCommentById(savedComment.getId()));
+    }
+
+    @Test
+    public void testGetAllCommentsForCourse() throws Exception {
+        when(mockedUserService.getAuthorizedUser()).thenReturn(createMockedUser());
+        courseCommentServiceUnderTest.addCommentForCourse("Good", COURSE_ID);
+        courseCommentServiceUnderTest.addCommentForCourse("Very interesting", COURSE_ID);
+        List<Comment> listOfCommentsForCourse = courseCommentServiceUnderTest.getAllCommentsForCourse(COURSE_ID);
+        assertEquals("Getting all comments for course.", 2, listOfCommentsForCourse.size());
+    }
+
+    @Test
+    public void testDeleteCommentById() throws Exception {
+        when(mockedUserService.getAuthorizedUser()).thenReturn(createMockedUser());
+        CourseComment savedComment = courseCommentServiceUnderTest.addCommentForCourse("Very interesting", COURSE_ID);
+        courseCommentServiceUnderTest.deleteCommentById(savedComment.getId());
+        CourseComment deletedComment = courseCommentServiceUnderTest.getCommentById(savedComment.getId());
```

Comment:
It would be better to use repository here directly.

Reply:
Ok, I will change it.